lib/datadog/core/configuration/option.rb

```diff
@@ -63,6 +84,10 @@ def context_exec(*args, &block)
         def context_eval(&block)
           @context.instance_eval(&block)
         end
+
+        # Used for testing
+        attr_reader :precedence_set
+        private :precedence_set
```

Comment:
I believe this could be rewritten to: 

```ruby
private
attr_reader :precedence_set
```

I think is idiomatic ruby

Reply:
I tried, but Rubocop won't let me, not even with (`AllowModifiersOnSymbols: true`, which is the default) https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Style/AccessModifierDeclarations